pubsublite/publish/metadata.go

```diff
@@ -19,28 +20,43 @@ import (
 	"strings"
 )
 
-// Metadata holds the results of a published message.
+// Metadata holds the result of publishing a message to the Pub/Sub Lite
+// service.
 type Metadata struct {
+	// The topic partition the message was published to.
 	Partition int
-	Offset    int64
+
+	// The offset the message was assigned.
+	Offset int64
 }
 
 func (m *Metadata) String() string {
 	return fmt.Sprintf("%d:%d", m.Partition, m.Offset)
 }
 
-// ParseMetadata converts a string obtained from Metadata.String()
-// back to Metadata.
-func ParseMetadata(input string) (*Metadata, error) {
-	parts := strings.Split(input, ":")
+// ParseMetadata converts the server message ID string of a pubsub.PublishResult
+// to Metadata.
+//
+// Example:
+//   result := publisher.Publish(ctx, &pubsub.Message{Data: []byte("payload")})
+//   id, err := result.Get(ctx)
+//   if err != nil {
+//     // TODO: Handle error.
+//   }
+//   metadata, err := publish.ParseMetadata(id)
+//   if err != nil {
+//     // TODO: Handle error.
+//   }
+func ParseMetadata(serverID string) (*Metadata, error) {
```

Comment:
Nit; is this really a serverID? Maybe messageID, but that's possibly overloaded. messageDestination, even? I am not sure, I don't have a great alternative.

Reply:
As discussed, renamed to just "id", for lack of better names.